org.eclipse.jdt.core/formatter/org/eclipse/jdt/core/formatter/DefaultCodeFormatterConstants.java

```diff
@@ -1185,7 +1185,7 @@ public class DefaultCodeFormatterConstants {
 	 * @see #PRESERVE_POSITIONS
 	 * @since 3.12
```

Comment:
`@since 3.12`

I fear it is too late, it is released few years ago, so anyone used this would be broken after this patch.

@mateusz-matela : WDYT?

Reply:
I agree, just changing it like that would cause more trouble than it's worth.
If we really care to fix the typo, we can handle the old key as deprecated - that would need changes in `DefaultCodeFormatterOptions.setDeprecatedOptions()` and probably also `ProfileVersionerCore` in jdt-ui.

Reply:
Actually, build errors suggest that either the API constant cannot be changed at all, or with major version bump to 4.0.0 (I just bumped into similar problem in https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/280)
So the constant with a typo would need to stay as `@Deprecated` in API, and a different constant created, and the formatter changed to use the new constant. I guess this would only add to the confusion, so I'd recommend leaving it and adding a comment that we are aware of the typo but it can't be fixed.

Reply:
Fair enough, when creating the PR I should have specified that I wasn’t sure how to handle it, as I’ve seen some projects allow breaking changes for typos, others not.

I reverted the change to the constant value and added a note to the Javadoc instead. Is that okay or did you intend it to be regular code comment?